deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java

```diff
@@ -62,6 +64,18 @@ private void replaceWithQuarkusTemplateFiles() {
                             authFileFolder(),
                             "AuthenticationPropagationHeadersFactory.java"));
 
+            Object quarkusGeneratorProperties = additionalProperties.get(QUARKUS_GENERATOR_NAME);
+
+            if (!(quarkusGeneratorProperties instanceof Map)) {
+                throw new IllegalStateException("Quarkus generator properties not found.");
+            }
+
+            Object openApiSpecId = ((Map<?, ?>) quarkusGeneratorProperties).get("openApiSpecId");
+
+            supportingFiles.add(
+                    new SupportingFile("auth/oauth2AuthenticationProvider.qute",
+                            authFileFolder(),
+                            openApiSpecId + "OAuth2AuthenticationProvider.java"));
```

Comment:
Can't we use a [`@Produce`](https://www.baeldung.com/java-ee-cdi#the-produces-annotation) instead?

Reply:
How would I set the [openApiSpecId](https://github.com/quarkiverse/quarkus-openapi-generator/pull/239/files#diff-39e483efc06c10d6b2354724417946d7ca29fe7d5b6e4db9b5cf2fd080ac8f4c) with a producer?

Reply:
Can't we make expose it in the CDI context? Or at least produce/calc it in runtime?

Reply:
Yeah maybe. I would need some time to try it.
Can I do this in a follow-up issue so that we can unlock https://github.com/kiegroup/kogito-runtimes/pull/2734 quickly.

Reply:
Sure, can you create an issue to track this down so we can prioritize for the next version?

Reply:
Sure.
https://github.com/quarkiverse/quarkus-openapi-generator/issues/242